docs/rules/strict.md

```diff
@@ -1,215 +1,231 @@
-# Strict Mode Directives (strict)
+# require or disallow strict mode directives (strict)
 
-A strict mode directive at the beginning of a script or function body enables strict mode semantics.
+A strict mode directive is a `"use strict"` literal at the beginning of a script or function body. It enables strict mode semantics.
 
-When used globally, the entire script, including all contained functions, are strict mode code:
+When a directive occurs in global scope, strict mode applies to the entire script:
 
 ```js
 "use strict";
+
+// strict mode
+
+function foo() {
+    // strict mode
+}
 ```
 
-It is also possible to specify function-level strict mode, such that strict mode applies only to the function in which the directive occurs:
+When a directive occurs at the beginning of a function body, strict mode applies only to that function, including all contained functions:
 
 ```js
 function foo() {
     "use strict";
-    return;
+    // strict mode
 }
 
-var bar = function() {
-    "use strict";
-    return;
+function foo2() {
+    // not strict mode
 };
+
+(function() {
+    "use strict";
+    function bar() {
+        // strict mode
+    }
+}());
 ```
 
-Unlike scripts, ECMAScript modules are always in strict mode. Strict mode directives in ECMAScript modules have no effect.
+In **Node.js** and the **CommonJS** module system, a hidden function wraps each module and limits the scope of a "global" strict mode directive.
+
+In **ECMAScript** modules, which always have strict mode semantics, the directives are unnecessary.
 
 ## Rule Details
 
-This rule is aimed at using strict mode directives effectively, and as such, will flag any unexpected uses or omissions of strict mode directives.
+This rule requires or disallows strict mode directives.
 
-## Options
+This rule disallows strict mode directives, no matter which option is specified, if ESLint configuration specifies either of the following as [parser options](../user-guide/configuring#specifying-parser-options):
+
+* `"sourceType": "module"` that is, files are **ECMAScript** modules
+* `"impliedStrict": true` property in the `ecmaFeatures` object
 
-There are four options for this rule:
+## Options
 
-* `"safe"` - require `"use strict"` globally when inside a module wrapper and in function scopes everywhere else. This is the default.
-* `"never"` - disallow `"use strict"`.
-* `"global"` - require `"use strict"` in the global scope.
-* `"function"` - require `"use strict"` in function scopes only.
+This rule has a string option:
 
-All strict mode directives are flagged as unnecessary if ECMAScript modules or implied strict mode are enabled (see [Specifying Parser Options](../user-guide/configuring#specifying-parser-options)). This behaviour does not depend on the rule options, but can be silenced by disabling this rule.
+* `"safe"` (default) corresponds either of the following options:
+    * `"global"` if ESLint considers a file to be a **Node.js** or **CommonJS** module
```

Comment:
We probably just want to say "CommonJS" because Node.js will support ES modules within the next year